src/Types/Type.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace ILIAS\Types;
+
+interface Type {
+
+	/**
+	 * @return string A string representation of the Type.
+	 */
+	function __toString();
+
+	/**
+	 * Is this type a subtype of $type. Not strict! x->isSubtype(x) == true.
```

Comment:
I second the comment of @Amstutz. What can I expect if one type is a subtype of another type? If this is intended to be understood in the "subclass" sense, the implementation of this method for Lists is incorrect, as explained [here](http://stackoverflow.com/a/2745301).